stdlib/format.mli

```diff
@@ -107,6 +103,446 @@
 
 *)
 
+(** {1 Tutorial}
+
+{2 Principles}
+
+Line breaking is based on three concepts:
+
+{ul {- {b boxes} : a box is a logical pretty-printing unit, which
+    defines a behaviour of the pretty-printing engine to display the
+    material inside the box.}
+    {- {b break hints}: a break hint is a directive to the
+      pretty-printing engine that proposes to break the line here, if it is
+      necessary to properly print the rest of the material.
+      Otherwise, the pretty-printing engine never break lines (except
+      "in case of emergency" to avoid very bad output).
+      In short, a break hint tells the pretty printer that a line break here
+      may be appropriate.}
+    {- {b indentation rules}:
+      When a line break occurs, the pretty-printing engines fixes the
+      indentation (or amount of leading spaces) of the new line using
+      indentation rules, as follows:
+    {ul {- A box can state the extra indentation of every new line opened in
+        its scope. This extra indentation is named
+        {b box breaking indentation}.}
+        {- A break hint can also set the additional indentation of the new line
+        it may fire. This extra indentation is named {b hint breaking
+        indentation}.}
+        {-  If break hint [bh] fires a new line within box
+            [b], then the indentation of the new line is simply the sum of:
+            the current indentation of box [b]
+            +
+            the additional box breaking indentation, as defined by box [b]
+            +
+            the additional hint breaking indentation, as defined by break
+            hint [bh].}}}}
+
+{2 Boxes}
+
+There are 4 types of boxes. (The most often used is the "hov" box type, so skip
+the rest at first reading).
+
+  - {b horizontal box} ({i h} box, as obtained by the
+    {!open_hbox} procedure): within this box, break hints do not
+    lead to line breaks.
+  - {b vertical box} ({i v} box, as obtained by the
+    {!open_vbox} procedure): within this box, every break hint lead
+    to a new line.
+  - {b vertical/horizontal box} ({i hv} box, as obtained by
+    the {!open_hvbox} procedure): if it is possible, the entire box
+    is written on a single line; otherwise, every break hint within the box
+    leads to a new line.
+  - {b vertical or horizontal box} ({i hov} box, as obtained
+    by the {!open_box} or {!open_hovbox} procedures): within this box, break
+    hints are used to cut the line when there is no more room on the line.
+    There are two kinds of "hov" boxes, you can find the details
+    below. In first approximation, let me
+    consider these two kinds of "hov" boxes as equivalent and
+    obtained by calling the {!open_box} procedure.
+
+Let me give an example. Suppose we can write 10 chars before
+the right margin (that indicates no more room). We represent any
+char as a [-] sign; characters [\[] and [\]]
```

Comment:
Rather than introducing a purely symbolic DSL, I would recommend to use format strings.
For instance, using `--@ --@ --` instead of `--b--b--` would make it easier to follow the tutorial in a toplevel.